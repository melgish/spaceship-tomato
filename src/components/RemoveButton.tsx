import type { ButtonHTMLAttributes } from 'react'

type Props = Readonly<Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type' | 'className'>>

export default function RemoveButton({ children = 'Remove', ...props }: Props) {
  return (
    <button
      {...props}
      className="px-1 py-0 text-red-800 font-bold border-red-800 text-sm"
      type="button"
    >
      {children}
    </button>
  )
}
