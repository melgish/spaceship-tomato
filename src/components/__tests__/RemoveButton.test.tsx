import { expect, it } from 'vitest'
import { render } from '@testing-library/react'
import RemoveButton from '@/components/RemoveButton'

it('should render', () => {
  const { container } = render(<RemoveButton />)

  expect(container).toMatchSnapshot()
})
