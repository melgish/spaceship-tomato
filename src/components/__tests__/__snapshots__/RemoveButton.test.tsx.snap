// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render 1`] = `
<div>
  <button
    class="px-1 py-0 text-red-800 font-bold border-red-800 text-sm"
    type="button"
  >
    Remove
  </button>
</div>
`;
